### PR TITLE
update LowerColorado with bmi and flowveldepth function implemented 

### PR DESCRIPTION
**Pull request by AminTorabi-NOAA**
A code added in LowerColorado_TX_v4 for running the bmi. It uses the "test_AnA_V4_HYFeature" yaml file.

## Additions
- **make sure in test_AnA_V4_HYFeature.yaml file lastobs_output_folder commented out ( : #lastobs/)**
- The flowveldepth functions added to model_DAforcing file.
- LOG also added instead of print (using log_level_set function)
- waterbodies_df was giving error in reconstructing the dataframe that some column are missing. I add the 3 columns of 'lon', 'lat', 'crs' and after reconstructing the dataframe drped them.
- flowveldepth function added
- run_all_configs_with_BMI_V2.py file is updated on the base_path address to be independent of my directory.
- output_parameters added in the troute_model file

## Removals

-

## Changes

-

## Testing

1. Tested on LowerColorado_TX_v4 and it works.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

**Review by shorvath-noaa — COMMENTED** [review]
Nice work. I've left a few comments to be addressed.

File: test/LowerColorado_TX_v4/test_AnA_V4_HYFeature.yaml
[the remark this answers is not in the document]
Comment (shorvath-noaa) [reply test/LowerColorado_TX_v4/test_AnA_V4_HYFeature.yaml:1]: @AminTorabi-NOAA actually, ignore this comment. Changing the output parameters to being on in the yaml file is good I think.

File: src/model_DAforcing.py
[the remark this answers is not in the document]
Comment (AminTorabi-NOAA) [reply src/model_DAforcing.py:1]: Good catch, Thanks. Done

File: test/LowerColorado_TX_v4/run_all_configs_with_BMI_V2.py
[the remark this answers is not in the document]
Comment (AminTorabi-NOAA) [reply test/LowerColorado_TX_v4/run_all_configs_with_BMI_V2.py:1]: Done

File: test/LowerColorado_TX_v4/run_all_configs_with_BMI_V2.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import sys
+import os
+import glob
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import pickle
+from datetime import datetime, timedelta
+import time
```
[the remark this answers is not in the document]
Comment (AminTorabi-NOAA) [reply test/LowerColorado_TX_v4/run_all_configs_with_BMI_V2.py]: Done

File: src/model_DAforcing.py [hunk 2]
```diff
@@ -316,12 +316,16 @@ def run(self, values: dict):
...
-        if output_parameters['lite_restart'] is not None:
-            _write_lite_restart(
-                values,
-                self._t0,
-                output_parameters['lite_restart']
-            )   
+        q0 = values.get('q0')
+        waterbody_df = values.get('waterbody_df')
+        lastobs_df = values.get('lastobs_df')
+        if len(q0) != 0 and len(waterbody_df) != 0 and len(lastobs_df) != 0:
```
[the remark this answers is not in the document]
Comment (AminTorabi-NOAA) [reply src/model_DAforcing.py]: Updated in a way that when `write_lite_restart` is 1 it go through `_write_lite_restart` function

File: src/model_DAforcing.py [hunk 3]
```diff
@@ -331,9 +335,20 @@ def run(self, values: dict):
                 lastobs_output_folder
             )
             
-        if output_parameters.get('stream_output',{}).get('qvd_nudge'):
-            #write flowveldepth file
-            pass
+        # if output_parameters.get('stream_output',{}) is not None and output_parameters.get('stream_output',{}).get('qvd_nudge'):
```
[the remark this answers is not in the document]
Comment (AminTorabi-NOAA) [reply src/model_DAforcing.py]: Done

**Review by shorvath-noaa — COMMENTED** [review]
In addition to the specific comments, I got the following error when I tried to run `run_with_BMI.py`:

```
***************test AnA Run***************
Initialize DA Module...
Traceback (most recent call last):
  File "/home/sean.horvath/projects/t-route/test/LowerColorado_TX_v4/run_with_BMI.py", line 132, in <module>
    run_troute('test AnA')
  File "/home/sean.horvath/projects/t-route/test/LowerColorado_TX_v4/run_with_BMI.py", line 38, in run_troute
    DAforcing.initialize(bmi_cfg_file = base_path + config_dict[config]['yaml'])
  File "/home/sean.horvath/projects/t-route/src/bmi_DAforcing.py", line 117, in initialize
    self._model = DAforcing_model(bmi_cfg_file) 
  File "/home/sean.horvath/projects/t-route/src/model_DAforcing.py", line 84, in __init__
    self._usgs_df.
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/pandas/core/frame.py", line 10535, in resample
    return super().resample(
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/pandas/core/generic.py", line 8312, in resample
    return get_resampler(
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/pandas/core/resample.py", line 1423, in get_resampler
    return tg._get_resampler(obj, kind=kind)
  File "/home/sean.horvath/miniconda-template/miniconda3/lib/python3.9/site-packages/pandas/core/resample.py", line 1599, in _get_resampler
    raise TypeError(
TypeError: Only valid with DatetimeIndex, TimedeltaIndex or PeriodIndex, but got an instance of 'Index'
```

**Pull request closed without merging**